model/task/task.go

```diff
@@ -179,6 +179,10 @@ type Task struct {
 	// CheckRunPath is a local file path to an output json file for the checkrun.
 	CheckRunPath string `bson:"check_run_path" json:"check_run_path"`
 
+	// CheckRunId is the id for the checkrun that was created in github.
+	// This is used to update the checkrun for future executions of the task.
+	CheckRunId *int `bson:"check_run_id,omitempty" json:"check_run_id,omitempty"`
```

Comment:
Can the check run ID be 0? If it's impossible for it to be 0, I feel like not using a pointer (and just checking `CheckRunId == 0` instead) might be slightly preferable to lots of nil checking when we're not sure if it's set.

Reply:
I don't know... 
I want to assume that not, but all the docs say is: 
<img width="454" alt="image" src="https://github.com/evergreen-ci/evergreen/assets/13104717/3ea8f7a7-4547-404d-9d42-b2f7d66eb2aa">

and they use the example of ` "id": 4,` in a response. So I don't feel like I can rely on them not being 0.